api/logs/src/main/java/io/opentelemetry/api/logs/DefaultLogger.java

```diff
@@ -12,22 +12,36 @@
 
 class DefaultLogger implements Logger {
 
-  private static final Logger INSTANCE = new DefaultLogger();
+  private static final Logger INSTANCE_WITH_DOMAIN = new DefaultLogger(/* hasDomain= */ true);
+  private static final Logger INSTANCE_NO_DOMAIN = new DefaultLogger(/* hasDomain= */ false);
 
-  private static final LogRecordBuilder NOOP_LOG_RECORD_BUILDER = new NoopLogRecordBuilder();
+  private static final EventBuilder NOOP_LOG_RECORD_BUILDER = new NoopLogRecordBuilder();
 
-  private DefaultLogger() {}
+  private final boolean hasDomain;
 
-  static Logger getInstance() {
-    return INSTANCE;
+  private DefaultLogger(boolean hasDomain) {
+    this.hasDomain = hasDomain;
+  }
+
+  static Logger getInstance(boolean hasDomain) {
+    return hasDomain ? INSTANCE_WITH_DOMAIN : INSTANCE_NO_DOMAIN;
+  }
+
+  @Override
+  public EventBuilder eventBuilder(String name) {
+    if (!hasDomain) {
+      throw new IllegalStateException(
+          "Cannot emit event from Logger without event domain. Please use LoggerBuilder#setEventDomain(String) when obtaining Logger.");
+    }
```

Comment:
is it allowed to throw exception in API? (vs log error and return noop)

Reply:
Logging and returning noop is more consistent with the other API level checks we have. 👍 

Will update.